src/components/clipboard/ClipboardButton.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from 'react';
+
+import { PiCopyLight } from 'react-icons/pi';
+import { TiTick } from 'react-icons/ti';
+import { Tooltip } from 'react-tooltip';
+
+import './ClipboardButton.scss';
+
+const TIMEOUT = 1500;
+const CLIPBOARD = 'clipboard';
+
+const copyTextToClipboard = async (text) => {
+  if (CLIPBOARD in navigator) {
+    return await navigator.clipboard.writeText(text);
+  }
+};
+
+type ClipboardButtonProps = {
+  text: string;
+};
+
+export const ClipboardButton = ({ text }: ClipboardButtonProps) => {
+  const [isCopied, setIsCopied] = useState(false);
+
+  const handleCopy = async (event) => {
+    event.stopPropagation();
+
+    copyTextToClipboard(text)
+      .then(() => {
```

Comment:
Was there a particular reason to use async and `.then()` rather than `await`?

Reply:
When I `await`ed the `copyTextToClipboard` function I got no response and using the `then` was how the example I used did it. I did this over a week ago now and I can't remember what source I used. 

Honestly, I didn't give it any real thought, since the `await` returned nothing and I only wanted to do the `then` bit if successful, I just used this method. I could have used a `try/catch` but I tend to find them overly verbose and ugly for async stuff, this just seemed that little bit neater to me.